src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -90,6 +90,23 @@ internal sealed class VariableInformation : IVariableInformation
         private string DisplayHint { get; set; }
         public bool IsPreformatted { get; set; }
 
+        public string Address()
+        {
+            // ask GDB to evaluate "&expression"
+            string command = "&("+_strippedName+")";
```

Comment:
>_strippedName [](start = 34, length = 13)

You need the `FullName()` instead of `_strippedName`

Reply:
A test case to try BTW: stick a pointer/reference in the watch window, expand it, use the feature on one of the child fields.

---
In reply to: [581388881](https://github.com/microsoft/MIEngine/pull/1107#discussion_r581388881) [](ancestors = 581388881)

Reply:
Break When Value Changes works when setting a data breakpoint from the Watch window
![image](https://user-images.githubusercontent.com/60245970/108929431-e50b6300-75f8-11eb-8565-1a695f6f7fdb.png)

Reply:
Edit: Unresolving since Break When Value Changes is breaking in the Watch window...

Reply:
Update: Changing from _strippedName to FullName() breaks Break When Value Changes in the Watch window. Would it be okay to leave this as is then, @gregg-miskelly?

Reply:
Do you mean root items in the watch window? If so, what are you seeing for `FullName()` and `_strippedName`?

Reply:
`FullName(): "((int *)0x7fff93033e0c)->*&testint"`
`_strippedName: "*&testint"`

Reply:
So that is a bug in our full name generation in this case, though I am not sure how easy it would be to fix. We can't just use _strippedName because that, in the general case, isn't an expression.

Ex:
struct S
{
   int myField;
};
S s = {};

Stick 's' in the watch window and expand it, try to set a data breakpoint on 'myField'. `_strippedName` will be `myField`.

Reply:
I *think* what we will need to do is special case children than start with `*` here: https://github.com/microsoft/MIEngine/blob/0e48a887feb19cf525025fa8bd5788464b667065/src/MIDebugEngine/Engine.Impl/Variables.cs#L272. We probably need a new NodeType of 'Dereference'.

I would test something like:

```
struct S2
{
     int *p;
};
struct S1
{
    S2 s2;
};
S1 s1 = {};
int myVar = 42;
s1.s2.p = &myVar;
```

Now put s1 in the watch window and expand all the way to '*p'. Also do the same but starting with 's1.s2' in the watch window. Assuming we wind up with a stripped name of '*p' in both cases then I think the way to fix this is to change the full name to be `*(<parent_name_here)`.

Reply:
Okay. Looking into this. Thanks.

Reply:
Update: I have verified the following:
- `_strippedName = "myField"` when sticking 's' in the Watch window, expanding it, and setting a data breakpoint on 'myField'
- `_strippedName = "*p"` when sticking both 's1' and 's1.s2' in the Watch window, expanding them, and setting data breakpoints on '*p'

Reply:
Hi @gregg-miskelly, I found this comment in the NodeType.Field case in FullName():
https://github.com/microsoft/MIEngine/blob/0e48a887feb19cf525025fa8bd5788464b667065/src/MIDebugEngine/Engine.Impl/Variables.cs#L120-L121

From what I understand, this may be the bug you are referring to as we need an expression for the commands in Address() and Size(). That said, I have a few questions:
1. Is it correct to say that this is why I was encountering the `Unable to set data breakpoint. One or more errors occurred.` error when switching from _strippedName to FullName()?
2. Should I be modeling NodeType.Dereference after NodeType.Field? It seems the new NodeType should be similar but not exactly the same since NodeType.Field has a conditional in `FullName()` that cuts off the expression and replaces it with an address instead.

Thanks!

Reply:
Its related, but not really the problem -- without that code, in the example I have above, you would see a full name for '*p' of `s1.s2.p.*p` instead of `*(s1.s2.p)`. The full name stuff is really a bit of a heuristic to guess the names that GDB gives us and try to turn that into what the full expression is. But that code doesn't handle deferencing a pointer.